projenrc/github-workflows.ts

```diff
@@ -493,3 +493,152 @@ export function runCommitLintWorkflow(project: AwsCdkConstructLibrary) {
     }
   }
 }
+
+/**
+ * Runs the code generation step to update the list of available JumpStart models, and DLC images.
+ * @param project AwsCdkConstructLibrary
+ */
+export function buildCodeGenerationWorkflow(project: AwsCdkConstructLibrary) {
+
+  const CREATE_PATCH_STEP_ID = 'create_patch';
+  const PATCH_CREATED_OUTPUT = 'patch_created';
+
+  // First job to run the code generation step
+  const generate: Job = {
+    name: 'Code generation',
+    runsOn: ['ubuntu-latest'],
+    permissions: {
+      contents: JobPermission.READ,
+    },
+    outputs: {
+      [PATCH_CREATED_OUTPUT]: {
+        stepId: CREATE_PATCH_STEP_ID,
+        outputName: PATCH_CREATED_OUTPUT,
+      },
+    },
+    steps: [
+      {
+        name: 'Checkout project',
+        uses: 'actions/checkout@v3',
+        with: {
+          ref: 'main',
+        },
+      },
+      {
+        name: 'Setup Node.js',
+        uses: 'actions/setup-node@1a4442cacd436585916779262731d5b162bc6ec7',
+        with: {
+          'node-version': '20.x',
+        },
+      },
+      {
+        name: 'Install dependencies',
+        run: 'yarn install --check-files --frozen-lockfile',
+      },
+      {
+        name: 'Run code generation',
+        run: 'npx projen generate-models-containers',
+      },
+      {
+        name: 'Find mutations',
+        id: CREATE_PATCH_STEP_ID,
+        run: [
+          'git add .',
```

Comment:
Consider a check that only the expected files have changed:
* apidocs/classes/JumpStartModel.md
* src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-model.ts
* src/patterns/gen-ai/aws-model-deployment-sagemaker/jumpstart-models.json

Reply:
Replicated the github workflow already available to upgrade main, which uses ```git add . as well```
More files can actually change, since the code generation step also updates the list of available DLC images
What about running it as is for some cycles and update both workflows with only the files we expect to change ? It will open a PR for review so we can catch any issue